sources/packages/backend/apps/api/src/services/institution/institution.service.ts

```diff
@@ -960,4 +953,27 @@ export class InstitutionService extends RecordDataModelService<Institution> {
       await this.institutionUserRepo.save(institutionUser);
     }
   }
+
+  /**
+   * Find if the institution
+   * has access to student data of given student.
+   * @param institutionId institution.
+   * @param studentId student.
+   * @returns value which specifies if the institution
+   * has access to student data of given student.
+   */
+  async hasStudentDataAccess(
+    institutionId: number,
+    studentId: number,
+  ): Promise<boolean> {
+    const institutionStudentDataAccess = await this.applicationRepo.findOne({
+      select: { id: true },
```

Comment:
The current Typeorm version that we are using has the exists now. Can we try to apply it here?
```ts
  return this.applicationRepo.exist({
      where: {
        student: { id: studentId },
        location: { institution: { id: institutionId } },
        applicationStatus: Not(ApplicationStatus.Overwritten),
      },
    });
```